commit e10a63b7e90009c38c9b4b6bf6823b441d57a1d9
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Fri Mar 9 17:28:07 2018 +0000

    lan78xx: Change LEDs to include 10Mb activity
    
    The default LED modes put 1000Mb/activity on orange and 100Mb/activity
    on green, but this leaves no indication for a 10Mb link. Change the
    defaults to put 10Mb/100Mb/activity on green.
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/drivers/net/usb/lan78xx.c b/drivers/net/usb/lan78xx.c
index 1b9f1c7ff609..8043a5a7f43a 100644
--- a/drivers/net/usb/lan78xx.c
+++ b/drivers/net/usb/lan78xx.c
@@ -2017,6 +2017,7 @@ static int lan78xx_phy_init(struct lan78xx_net *dev)
 {
 	int ret;
 	u32 mii_adv;
+	u32 led_modes;
 	struct phy_device *phydev;
 
 	phydev = phy_find_first(dev->mdiobus);
@@ -2089,6 +2090,19 @@ static int lan78xx_phy_init(struct lan78xx_net *dev)
 	mii_adv = (u32)mii_advertise_flowctrl(dev->fc_request_control);
 	phydev->advertising |= mii_adv_to_ethtool_adv_t(mii_adv);
 
+	/* Change LED defaults:
+	 *   orange = link1000/activity
+	 *   green  = link10/link100/activity
+	 * led: 0=link/activity          1=link1000/activity
+	 *      2=link100/activity       3=link10/activity
+	 *      4=link100/1000/activity  5=link10/1000/activity
+	 *      6=link10/100/activity    14=off    15=on
+	 */
+	led_modes = phy_read(phydev, 0x1d);
+	led_modes &= ~0xff;
+	led_modes |= (1 << 0) | (6 << 4);
+	(void)phy_write(phydev, 0x1d, led_modes);
+
 	genphy_config_aneg(phydev);
 
 	dev->fc_autoneg = phydev->autoneg;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1141_e10a63b7e90009c38c9b4b6bf6823b441d57a1d9.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1141_e10a63b7e90009c38c9b4b6bf6823b441d57a1d9.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
